posthog/models/sessions/sql.py

```diff
@@ -0,0 +1,130 @@
+from django.conf import settings
+
+from posthog.clickhouse.table_engines import (
+    Distributed,
+    ReplicationScheme,
+    AggregatingMergeTree,
+)
+
+TABLE_BASE_NAME = "sessions"
+SESSIONS_DATA_TABLE = lambda: f"sharded_{TABLE_BASE_NAME}"
+
+# if updating these column definitions
+# you'll need to update the explicit column definitions in the materialized view creation statement below
+SESSIONS_TABLE_BASE_SQL = """
+CREATE TABLE IF NOT EXISTS {table_name} ON CLUSTER '{cluster}'
+(
+    -- part of order by so will aggregate correctly
+    session_id VARCHAR,
+    -- part of order by so will aggregate correctly
+    team_id Int64,
+    -- ClickHouse will pick any value of distinct_id for the session
+    -- this is fine since even if the distinct_id changes during a session
+    -- it will still (or should still) map to the same person
+    distinct_id SimpleAggregateFunction(any, String),
+    min_first_timestamp SimpleAggregateFunction(min, DateTime64(6, 'UTC')),
+    max_last_timestamp SimpleAggregateFunction(max, DateTime64(6, 'UTC')),
+
+    urls SimpleAggregateFunction(groupUniqArrayArray, Array(String)),
+    entry_url AggregateFunction(argMin, String, DateTime64(6, 'UTC')),
+    exit_url AggregateFunction(argMax, String, DateTime64(6, 'UTC')),
+    initial_utm_source AggregateFunction(argMin, String, DateTime64(6, 'UTC')),
+    initial_utm_campaign AggregateFunction(argMin, String, DateTime64(6, 'UTC')),
+    initial_utm_medium AggregateFunction(argMin, String, DateTime64(6, 'UTC')),
+    initial_utm_term AggregateFunction(argMin, String, DateTime64(6, 'UTC')),
+    initial_utm_content AggregateFunction(argMin, String, DateTime64(6, 'UTC')),
+    initial_referring_domain AggregateFunction(argMin, String, DateTime64(6, 'UTC')),
+
+    event_count SimpleAggregateFunction(sum, Int64),
+    pageview_count SimpleAggregateFunction(sum, Int64),
```

Comment:
Let's include autocapture count as well, useful for calculating whether a session is a bounce

Reply:
ClickHouse has a map type... I'm not sure how to use it but could conceivably have `Map<event_name, int>` instead of individuals

(standard disclaimer: it's something that exists, not something I've used 🤣)

https://clickhouse.com/blog/aggregate-functions-combinators-in-clickhouse-for-arrays-maps-and-states#aggregating-maps

Reply:
Yeah, this would be really good if we can get this working

Reply:
I got this working with a sumMap, so yeah let's include it. I also want to keep the specific columns for pageviews and autocaptures, as they're used a lot for some important queries so let's keep them fast.